src/xunit.analyzers.tests/Analyzers/X1000/MemberDataShouldReferenceValidMemberTests.cs

```diff
@@ -837,6 +837,55 @@ public class TestClass {{
 		await Verify.VerifyAnalyzer(LanguageVersion.CSharp10, source);
 	}
 
+	[Fact]
+	public async void DoesNotFindWarning_WithIntArrayArguments()
+	{
+		var source = $@"
+using System.Collections.Generic;
+using Xunit;
+
+public class TestClass
+{{
+  public static IEnumerable<object[]> GetSequences(IEnumerable<int> seq) =>
+    new List<object[]>();
+
+  [Theory]
+  [MemberData(nameof(GetSequences), new int[] {{ 1, 2 }})]
+  [MemberData(nameof(GetSequences), new [] {{ 3, 4, 5}})]
+  public void Test(IEnumerable<int> seq)
+  {{
+    Assert.NotEmpty(seq);
+  }}
+}}
+";
+
+		await Verify.VerifyAnalyzer(LanguageVersion.CSharp10, source);
+	}
+
+	[Fact]
+	public async void DoesNotFindWarning_WithObjectArrayArguments()
+	{
+		var source = $@"
+using System.Collections.Generic;
+using Xunit;
+
+public class TestClass
+{{
+  public static IEnumerable<object[]> GetSequences(IEnumerable<object> seq) =>
+    new List<object[]>();
+
+  [Theory]
+  [MemberData(nameof(GetSequences), new object[] {{ 1, 2 }})]
```

Comment:
Technically this case _should_ have a warning. The `int[]` does not have warning, because it becomes the first parameter passed to a `params object[] parameters`. However, an `object[]` binds tighter than the `params`, so this will actually pass the parameters `1` and `2` to the method instead of `object[] { 1, 2 }`. 

NB: Confirmed via testing - top method here works successfully, but with the xUnit warnings; but this version fails with an `ArgumentException`.

Reply:
@viceroypenguin Gotcha, will try another approach

Reply:
@viceroypenguin This one should work